web/src/installers/installer-versions.ts

```diff
@@ -4,14 +4,41 @@ import Bugsnag from "@bugsnag/js";
 import { HTTPError } from "../server/errors";
 import { getPackageUrl } from "../util/package";
 import { InstallerVersions } from "./versions";
+import * as semver from "semver";
+import {logger} from "../logger";
 
 interface IInstallerVersions {
   [addon: string]: string[];
 }
 
 const installerVersionsCache: { [url: string]: IInstallerVersions } = {};
+let externalAddons = {};
+let externalAddonTimer;
 
-export async function getInstallerVersions(distUrl: string, kurlVersion?: string): Promise<IInstallerVersions> {
+function mergeAddonVersions(internalAddonVersions: IInstallerVersions, kurlVersion?: string) {
+  const addons: IInstallerVersions = {};
+  Object.keys(externalAddons).forEach(externalAddonName => {
+    const fileName = externalAddonName.slice(0, externalAddonName.length - 7);
+    const [name, version] = fileName.split("-");
```

Comment:
Can we guarantee there are no dashes in any addon filename nor will ever be?

Reply:
Since we're importing the addon, we can make this always the case but I would need to tweak the import process, which might be good.  Since we're only looking to do this for kots right now I've been keeping things more simplistic.